packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -405,11 +434,26 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<
     }
   }, [rangePosition, value]); // eslint-disable-line
 
+  const applyBoundaries = (contiguousRangeBoundary, defaultBoundary) => {
+    if (availableRangePositions.length === 1 && availableRangePositions[0] === 'start') {
```

Comment:
Is this verification enough to assure we don't update the boundaries when selecting the `start` date on Date**Time**Range?